astropy/cosmology/flrw/base.py

```diff
@@ -1505,7 +1512,7 @@ def nonflat(self: _FlatFLRWMixinT) -> _FLRWT:
         # Make new instance, respecting args vs kwargs
         inst = self.__nonflatclass__(*ba.args, **ba.kwargs)
         # Because of machine precision, make sure parameters exactly match
-        for n in inst.__all_parameters__ + ("Ok0",):
+        for n in chain(inst._parameters_all, ("Ok0",)):
```

Comment:
Here I think `inst._parameters_all + {"Ok0"}` is just better than `chain`

Reply:
Should this be `_parameters`? I.e., exclude derived ones?

Reply:
When moving to non-flat classes we need to include `Ok0`.

Reply:
I meant whether it was correct to include the derived parameters; adding "Ok0" makes sense.